### PR TITLE
Move `WorkerPool` to `execution`

### PR DESCRIPTION
**Pull request by TimDiekmann**
This is part of [Better separation of crates](https://app.asana.com/0/1201707629991362/1201481007343142/f) (_internal_).

## 🌟 What is the purpose of this PR?

Renames `WorkerPoolController` to `WorkerPool` and moves it to `execution`

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202055186269230/1202099498122333/f) _(internal)_

## 🚫 Blocked by

- #534

## ⚠️ Known issues

The file `worker_pool/mod.rs` already existed, so GitHub isn't able to detect the movement of the file.

## 🛡 What tests cover this?

The integration test suite covers this

### CODE REVIEW

File: packages/engine/lib/execution/src/runner/mod.rs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 //! Language runner implementations to run [`package`]s.
 //!
-//! Currently, three [`Language`] runners are available: [`JavaScriptRunner`], [`PythonRunner`], and
-//! [`RustRunner`]. The latter is currently disabled. To drive the language runners, the [`comms`]
-//! module provides messages to be sent to the runners or received from the runners.
+//! Currently, three [`Language`] runners are available: JavaScript, Python, and Rust. The latter
+//! is currently disabled due to an incomplete implementation. To drive the language runners, the
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/runner/mod.rs:4]:
```suggestion
//! Currently, three [`Language`] runners are available: JavaScript, Python, and Rust. The latter
//! is WIP and currently disabled. To drive the language runners, the
```

Comment (Alfred-Mountfield) [reply packages/engine/lib/execution/src/runner/mod.rs:4]: (probably messed up the comment formatting)

File: packages/engine/lib/execution/src/worker/mod.rs [hunk 2]
```diff
@@ -1,9 +1,5 @@
-//! The `worker` module defines the three different language runners for JavaScript, Python, and
-//! Rust and the accompanying API for their tasks and communication.
-//!
-//! The [`runner`] module contains the implementations for the languages and the
-//! [communication module](runner::comms). The [`task`] module defines tasks executed in the
-//! runners.
+//! Defines the three different language runners for JavaScript, Python, and Rust and the
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/worker/mod.rs:1]: Maybe defines the ``[`Worker`]`` struct and the three language runners contained within it (maybe with doc links)

File: packages/engine/lib/execution/src/worker_pool/comms/mod.rs [hunk 3]
```diff
@@ -1,3 +1,16 @@
+//! Communication channels and messages for running the [`WorkerPool`].
+//!
+//! This module contains channals and messages for different purposes:
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/worker_pool/comms/mod.rs:3]:
```suggestion
//! This module contains channels and messages for different purposes:
```

File: packages/engine/lib/execution/src/worker_pool/mod.rs [hunk 4]
```diff
@@ -1,9 +1,457 @@
+//! Handling multiple workers simultaneously.
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/worker_pool/mod.rs:1]: Maybe something like

```suggestion
//! The [`WorkerPool`] manages [`Workers`] and the orchestration of [`Task`]s executed on them.
```

File: packages/engine/lib/execution/src/worker_pool/mod.rs [hunk 5]
```diff
@@ -1,9 +1,457 @@
...
+pub(crate) use self::{config::SplitConfig, handler::WorkerPoolHandler};
+use crate::{
+    package::PackageTask,
+    runner::comms::{ExperimentInitRunnerMsg, ExperimentInitRunnerMsgBase, NewSimulationRun},
+    task::{Task, TaskDistributionConfig, TaskId, TaskSharedStore},
+    worker::{SyncPayload, Worker, WorkerConfig, WorkerTask},
+    Error, Result,
+};
+
+/// Maintains multiple [`Worker`]s.
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/worker_pool/mod.rs:52]:
```suggestion
/// Manages multiple [`Worker`]s.
```

Comment (Alfred-Mountfield) [reply packages/engine/lib/execution/src/worker_pool/mod.rs:52]: Or perhaps orchestrates

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]

File: packages/engine/lib/execution/src/worker/mod.rs (hunk 2 above)
Comment (TimDiekmann) [reply packages/engine/lib/execution/src/worker/mod.rs:1]: The language runners are not defined here anymore. I changed the documentation to this:
```suggestion
//! Defines the task worker and accompanying API.
//!
//! The [`Worker`] contains and orchestrates the [language runners].
//! 
//! [language runners]: crate::runner
```

**Review by Alfred-Mountfield — APPROVED** [review]
🚀

**Pull request merged**